src/content/questionnaire/sections/B.tsx

```diff
@@ -314,8 +292,12 @@ const FormSectionB: FC<FormSectionProps> = ({ SectionOption, refs }: FormSection
         <SelectInput
           id="section-b-program"
           label="Program"
-          options={programOptions?.map(programToSelectOption)}
-          value={programOption?.name}
+          name="program[_id]"
```

Comment:
I noticed two issues with the selection field, I think they're both related:

1) If you select the "Other" option, and view the form in Read Only mode (e.g. Federal Lead or if you submit), when you try to reload the page, you get the unsaved changes alert and the field is outlined in red.


<img width="1112" alt="Screenshot 2024-12-12 at 10 09 32 AM" src="https://github.com/user-attachments/assets/e2024bfc-d7b4-4c7f-8c49-970294f88776" />

---

2) When you select the "Other" option, then move on to the next sections, the "Other" no longer persists in the dropdown. I like that you cleaned up all the other unnecessary properties stored in the program object, so without adding those back, do you think there's a way we can safely infer if the Other option was selected?

<img width="1112" alt="Screenshot 2024-12-12 at 10 17 19 AM" src="https://github.com/user-attachments/assets/cf34bb1d-4f87-499a-b917-d668155cb38b" />

<img width="1112" alt="Screenshot 2024-12-12 at 10 18 18 AM" src="https://github.com/user-attachments/assets/4a7864a5-77ae-4570-a804-56a7f725663e" />

Reply:
Thanks for catching this. This was a big oversight by me. My new implementation was not backwards compatible due to the removing of properties. I have added support for older forms in [9cbbfc3](https://github.com/CBIIT/crdc-datahub-ui/pull/562/commits/9cbbfc3074b4da1785ab0499a34ba48aa48dc706).

In order to differentiate from an empty form with "Other" vs "Not Applicable", I had to add a check for the existence of the old property "notApplicable". This was the only way to fully support older forms. LMK if you can think of a better way.

Please re-test and verify this for me as there are a lot of possible scenarios in this form.

Reply:
Additional note, for some reason if you try saving a brand new, un-edited, submission then the select would wanna default to the first option in the select, in this case being the NotApplicableProgram. Fixed this by adding an empty option in the ProxySelect. [cff287b](https://github.com/CBIIT/crdc-datahub-ui/pull/562/commits/cff287bd4a965fcfe0dc9a54f23571071655c4c0).

Reply:
Will let you resolve this if you feel it resolved the issues you described.